### PR TITLE
Charmhub libraries page - layout improvements

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done
- Ensure the layout uses the available screen space optimally at all screen sizes
- Small refactor to reduce the heavy use of class names on many elements, as a first step towards making it a modular component based page. 

## How to QA
- Visit the demo in your browser, link commented below
- Double check the flask conditionals, ensure the logic is sound
- Resize the screen, check at various resolutions
 
## Issue / Card
Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3637 (on the vanilla repo)

## Screenshots
image

### CODE REVIEW

**Comment by hatched** [comment]
Very nice looking 👍

File: static/sass/_pattern_p-docstring.scss [hunk 1]
```diff
@@ -1,24 +1,134 @@
...
+
+    & &:last-child {
+      padding-bottom: 0;
+    }
+
+    &__header {
+      @extend %bottom-section-padding--x-shallow;
+    }
+
+    // &__group:not(:last-child) {
```
Comment (solazio) [inline static/sass/_pattern_p-docstring.scss:67]: Is there any reason why this is commented out?

File: templates/details/libraries/docstring.html [hunk 2]
```diff
@@ -57,280 +60,245 @@ <h4 id="{{ group['name'].lower() }}" class="p-code u-no-margin--bottom">
...
-                                      {% if attribute.name %}
-                                      <span class="token keyword">{{ attribute.name }}</span>
-                                      {% endif %}
-                                      {% if attribute.type_name %}
-                                      <span class="token comment">({{ attribute.type_name }})</span>
-                                      {% endif %}
-                                    </p>
-                                  </div>
+              <div class="col-medium-5 col-6 col-x-large-7">
+              {% set count = namespace(value=1) %}
```
Comment (solazio) [inline templates/details/libraries/docstring.html:165]: You don't need this anymore.

**Review by solazio — APPROVED** [review]
Looks good, just a few small comments in the code.

File: static/sass/_pattern_p-docstring.scss (hunk 1 above)
Comment (lyubomir-popov) [reply static/sass/_pattern_p-docstring.scss:67]: no, deleted

**Pull request merged**